### PR TITLE
[CIAPP-3593] Explain tags query param in instructions for gitlab

### PR DESCRIPTION
**Pull request by AdrianLC**
### What does this PR do?

Explain how to provide custom tags to CI Visibility for pipelines. This is supported through a new query parameter in the webhooks.

Some integrations will receive updates in the UI. For now GitLab which is under review in https://gitlab.com/gitlab-org/gitlab/-/merge_requests/79665. We will request similar changes to Buildkite.

This PR documents how to use the new parameter with webhooks as well as the new input in the GitLab settings.

### Preview

- https://docs-staging.datadoghq.com/adrian.lopezcalvo%2Fci-visibility-gitlab-custom-tags/continuous_integration/setup_pipelines/gitlab/

### Additional Notes

**Do not merge** before https://gitlab.com/gitlab-org/gitlab/-/merge_requests/79665

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/continuous_integration/setup_pipelines/gitlab.md [hunk 1]
```diff
@@ -119,6 +123,8 @@ Go to **Settings > Webhooks** in your repository (or GitLab instance settings),
 
 To set custom `env` or `service` parameters, add more query parameters in the webhooks URL: `&env=<YOUR_ENV>&service=<YOUR_SERVICE_NAME>`
 
+To set custom tags, provide the query parameter `tags`, with *key:value* pairs separated by commas. If the tag  contains commas, quotes are needed. Example: `key1:value1,"key2: value with , comma",key3:value3`. (**This parameter needs URL encoding**, for the example: `&tags=key1%3Avalue1%2C%22key2%3A+value+with+%2C+comma%22%2Ckey3%3Avalue3`)
```
Comment (juan-fernandez) [inline content/en/continuous_integration/setup_pipelines/gitlab.md:126]:
```suggestion
To set custom tags, provide the query parameter `tags`, with *key:value* pairs separated by commas. If the tag contains commas, quotes are needed. Example: `key1:value1,"key2: value with , comma",key3:value3`. (**This parameter needs URL encoding**, for the example: `&tags=key1%3Avalue1%2C%22key2%3A+value+with+%2C+comma%22%2Ckey3%3Avalue3`)
```

**Review by juan-fernandez — APPROVED** [review]

File: content/en/continuous_integration/setup_pipelines/circleci.md [hunk 2]
```diff
@@ -29,6 +29,8 @@ The Datadog integration for [CircleCI][1] works by using [webhooks][2] to send d
 
 2. Click **Add Webhook** to save the new webhook.
 
+To set custom tags, add to the **Webhook URL** a query parameter `tags`, with *key:value* pairs separated by commas. If the tag contains commas, quotes are needed. Example: `key1:value1,"key2: value with , comma",key3:value3`. (**This parameter needs URL encoding**, for the example: `?tags=key1%3Avalue1%2C%22key2%3A+value+with+%2C+comma%22%2Ckey3%3Avalue3`)
```
Comment (fermayo) [inline content/en/continuous_integration/setup_pipelines/circleci.md:32]:
```suggestion
To set custom tags, add to the **Webhook URL** a URL-encoded query parameter `tags`, with `key:value` pairs separated by commas. If the tag pair contains any commas, quotes are needed. For example, to add `key1:value1,"key2: value with , comma",key3:value3`, the following string would need to be appended to the 
**Webhook URL**: `?tags=key1%3Avalue1%2C%22key2%3A+value+with+%2C+comma%22%2Ckey3%3Avalue3`
```

Maybe this reads better? I felt like the URL encoding should be raised at the beginning

File: content/en/continuous_integration/setup_pipelines/gitlab.md [hunk 3]
```diff
@@ -103,6 +103,10 @@ Fill in the integration configuration settings:
 : Specifies which environment (`env` tag) to attach to each span generated by the integration. Use this to differentiate between groups of GitLab instances (for example: staging or production).<br/>
 **Default**: `none`
 
+**Tags** (optional)
+: Specifies any custom tags to attach to each span generated by the integration. Provide one tag per line in the format: *key:value*.<br/>
```
Comment (fermayo) [inline content/en/continuous_integration/setup_pipelines/gitlab.md:107]:
```suggestion
: Specifies any custom tags to attach to each span generated by the integration. Provide one tag per line in the format: `key:value`.<br/>
```

File: content/en/continuous_integration/setup_pipelines/gitlab.md [hunk 4]
```diff
@@ -119,6 +123,8 @@ Go to **Settings > Webhooks** in your repository (or GitLab instance settings),
 
 To set custom `env` or `service` parameters, add more query parameters in the webhooks URL: `&env=<YOUR_ENV>&service=<YOUR_SERVICE_NAME>`
 
+To set custom tags, provide the query parameter `tags`, with *key:value* pairs separated by commas. If the tag contains commas, quotes are needed. Example: `key1:value1,"key2: value with , comma",key3:value3`. (**This parameter needs URL encoding**, for the example: `&tags=key1%3Avalue1%2C%22key2%3A+value+with+%2C+comma%22%2Ckey3%3Avalue3`)
```
Comment (fermayo) [inline content/en/continuous_integration/setup_pipelines/gitlab.md:126]:
```suggestion
To set custom tags, add to the **Webhook URL** a URL-encoded query parameter `tags`, with `key:value` pairs separated by commas. If the tag pair contains any commas, quotes are needed. For example, to add `key1:value1,"key2: value with , comma",key3:value3`, the following string would need to be appended to the 
**Webhook URL**: `?tags=key1%3Avalue1%2C%22key2%3A+value+with+%2C+comma%22%2Ckey3%3Avalue3`
```

**Review by fermayo — APPROVED** [review]

**Comment by fermayo** [comment]
> I'm not sure if it's being tracked somewhere else, but we should also surface this in the in-app onboarding.

For GitLab, we are tracking that in https://datadoghq.atlassian.net/browse/CIAPP-3622

> We should also add it to the in-app onboarding for CircleCI/BuildKite if and when they're supported. But maybe since it's only supported through the more convoluted webhook method, we should mention that they can add Tags, and point out to the docs instead of trying to explain within our in-app onboarding.

I've created https://datadoghq.atlassian.net/browse/CIAPP-3678 for CircleCI and https://datadoghq.atlassian.net/browse/CIAPP-3679 for Buildkite. Buildkite should update their UI to include something similar to GitLab.

**Review by jtappa — APPROVED** [review]

**Pull request merged**